wallet-new/server/Main.hs

```diff
@@ -123,16 +122,16 @@ actionWithNewWallet sscParams nodeParams params =
     mainAction
         :: PassiveWalletLayer Production
         -> NodeResources ext
-        -> (ActionSpec Kernel.Mode.WalletMode (), OutSpecs)
+        -> (Diffusion Kernel.Mode.WalletMode -> Kernel.Mode.WalletMode ())
     mainAction w nr = runNodeWithInit w nr
 
     runNodeWithInit
         :: PassiveWalletLayer Production
         -> NodeResources ext
-        -> (ActionSpec Kernel.Mode.WalletMode (), OutSpecs)
+        -> (Diffusion Kernel.Mode.WalletMode -> Kernel.Mode.WalletMode ())
     runNodeWithInit w nr =
-        let (ActionSpec f, outs) = runNode nr (plugins w)
-         in (ActionSpec $ \s -> f s, outs)
+        let f = runNode nr (plugins w)
+         in \s -> f s
```

Comment:
this one can actually be simplified to `runNodeWithInit w nr = runNode nr (plugins w)`